### DISCUSSION TITLE
How to delete Highlights?

### DISCUSSION THREAD

**Discussion by net-frog** [RyotaUshio/obsidian-pdf-plus#187] [Q&A]
Hi,

How to delete the Highlights from the PDF itself. I know you can open the .md note file and delete it from there, but it would be helpful if we could delete the highlights from the PDF itself.

Additionally, If you can share what is the workflow to edit or modify the highlight?

Thanks

**Comment by N3C2L** [comment]
If the highlight is written to the file directly: right click and delete annotation.
But I think you mean to delete the highlight if it's only linked and not written to the PDF itself.

1. The fastest way is to double click the highlight inside the PDF to select the link + text in the annotation .md file
2. Press esc to select the current line - possible with this plugin: https://github.com/gokulk16/select-current-line-plugin
3. Press the delete key on your keyboard

If you have no paragraph between your link and the text selection in the copy link format, then this steps above would delete the whole linked text selection and the link itself from the annotation file.

**Comment by tprotopopescu** [comment]
What about on mobile? I cannot work out how to delete annotations written to a pdf on mobile. On desktop either right clicking and using the context menu, or clicking and using the annotation popup are ways to do it. But on mobile there is no delete button on the annotation popup:
![Screenshot_20240605_120213_Obsidian](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/24751076/b4ae6687-3220-4f80-a277-d6c1a33561d1)


and there is no analogue of a right-click as far as I have been able to work out.

**Reply by RyotaUshio** [reply]
Thank you, this is a bug and I'm going to fix it in the next release.

**Reply by RyotaUshio** [reply]
For the meantime, you can show the context menu by long-pressing the annotation and then selecting "Delete annotation"

**Reply by tprotopopescu** [reply]
Thank you for that! I was not able to figure out why long pressing did not work for me, then I realised that you need to enable the pdf++ custom menu. However, the custom menu makes selecting text on mobile kind of difficult. As soon as a word is selected, by long pressing, the menu pops up and covers the text and you have to dismiss the menu before you can continue with the selection. 


https://github.com/RyotaUshio/obsidian-pdf-plus/assets/24751076/3df05f67-212f-4164-a481-efc2e5200a4f

**Reply by RyotaUshio** [reply]
Thanks! This will be also fixed in the next version. If you have more to report about the mobile experience, please let me know via #217.
